ingeniamotion/pdo.py

```diff
@@ -18,6 +21,130 @@
     from ingeniamotion.motion_controller import MotionController
 
 
+class PDOPoller:
+    """Poll register values using PDOs"""
+
+    def __init__(
+        self,
+        mc: "MotionController",
+        servo: str,
+        refresh_time: float,
+        buffer_size: int,
+    ) -> None:
+        """Constructor.
+
+        Args:
+            mc: MotionController instance
+            servo: drive alias.
+            refresh_time: PDO values refresh time.
+            buffer_size: Maximum number of register readings to store.
+
+        """
+        super().__init__()
+        self.__mc = mc
+        self.__servo = servo
+        self.__refresh_time = refresh_time
+        self.__buffer_size = buffer_size
+        self.__num_channels: int = 0
+        self.__buffer: Dict[int, Deque[Union[int, float]]] = defaultdict(partial(deque, maxlen=self.__buffer_size))  # type: ignore
```

Comment:
Why do you use `defaultdict`? I think this should be a list, as in the Poller, and it should be initialized in the start function.

Reply:
I don't think it makes a difference if it's a list or a dictionary. Both pollers don't need the same internals, as long as they work similarly.

Reply:
I think we should avoid the use `# type: ignore` if it is not for a good reason.

Reply:
Done